src/native/minipal/getexepath.h

```diff
@@ -55,6 +57,35 @@ static inline char* minipal_getexepath(void)
     }
 
     return strdup(path);
+#elif defined(__HAIKU__)
+    size_t bufferSize = 128;
```

Comment:
I would recommend using `char buffer[B_PATH_NAME_LENGTH];` (StorageDefs.h) and `strdup` like the other platforms.

Reply:
Oh, wasn't aware that Haiku also had a path limit.

Reply:
Done.